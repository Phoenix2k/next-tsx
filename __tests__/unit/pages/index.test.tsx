/// <reference types="jest"/>

import { shallow } from 'enzyme';
import { NextRouter } from 'next/router';
import Renderer from 'react-test-renderer';
import Home from '../../../src/pages/index';

jest.mock('next/router', () => ({
  useRouter(): Partial<NextRouter> {
    return {
      route: '/home'
    };
  }
}));

describe('Home page', () => {
  it('renders "Home page" heading', () => {
    const shallowRender = shallow(<Home />);
    expect(shallowRender.find('h1').text()).toEqual('Home page');
  });
  it('matches snapshot', () => {
    const snapshotRender = Renderer.create(<Home />);
    const tree = snapshotRender.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
