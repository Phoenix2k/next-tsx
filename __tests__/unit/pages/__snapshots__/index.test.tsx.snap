// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
.emotion-1 {
  border: 1px solid #ccc;
  margin: 1em;
  padding: 1em;
}

.emotion-0 .nav-link {
  color: blue;
}

.emotion-0 .nav-link--active {
  color: green;
}

.emotion-0 .nav-list li {
  display: inline-block;
}

.emotion-0 .nav-list ul {
  list-style: none;
  padding: 0;
}

<div
  className="emotion-1"
>
  <header>
    Header
  </header>
  <nav
    className="emotion-0"
  >
    <ul>
      <li>
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
    </ul>
  </nav>
  <main>
    <h1
      itemProp="title"
    >
      Home page
    </h1>
  </main>
  <footer>
    Footer
  </footer>
</div>
`;
